compile.ts

```diff
@@ -603,9 +603,88 @@ class Compiler {
       }
       return dest;
     } else if (ts.isStringLiteral(e)) {
-      return new Variable(new LiteralValue({ id: e.text }));
+      const value = new LiteralValue({ id: e.text });
+      if (dest) {
+        this.#emit(
+          methods.setReg,
+          value,
+          this.ref(dest, VariableOperations.Write)
+        );
+      } else {
+        return new Variable(value);
+      }
+      return dest;
+    } else if (ts.isArrayLiteralExpression(e)) {
+      let arr: number[] = [];
+      for(const property of e.elements) {
```

Comment:
Why is this needed when there's the `coord` function?
If we do need it we should probably check the array length. I could certainly see people trying to use general arrays and getting confused.

Reply:
I implemented it because I saw the typescript definition `type Coord = [number, number]` which implies a user should be allowed to write this when specifying a coordinate. But I'd be happy to just get rid of it, the `coord` function is better ergonomics IMO.

Reply:
Maybe change the `Coord` type to `{x: number, y: number}`, and allow this to be written literally? Just like `{id: "metalore", num: 1}` is also allowed to be written literally?

Or disallow writing both literally, and require usage of the `coord` and `value` functions? Then the type could change to:

```
class Coord {
  x: number;
  y: number
}
```

Reply:
Ah, I was only intending that to be used for arguments to functions which take a coordinate so you could enter a literal. You're right, an `{x:, y:}` object would probably be more clear.